drivers/sensor/isl29125/isl29125.c

```diff
@@ -0,0 +1,205 @@
+	}
+}
+
+static void isl29125_write16(struct device *i2c_dev, u8_t reg, u16_t data)
+{
+	u8_t buf[3];
+
+	data = sys_cpu_to_le16(data);
+	buf[0] = reg;
+	buf[1] = data & 0xFF;
+	buf[2] = data >> 8;
+	if (i2c_write(i2c_dev, buf, 3, ISL29125_I2C_ADDR)) {
+		LOG_ERR("Error writting register %d at Addr:0x%x", reg, ISL29125_I2C_ADDR);
+	}
+}
+
+static u16_t isl29125_readRed(struct device *i2c_dev)
+{
+	return isl29125_read16(i2c_dev, ISL29125_RED_L);
+}
+
+static u16_t isl29125_readGreen(struct device *i2c_dev)
+{
+	return isl29125_read16(i2c_dev, ISL29125_GREEN_L);
+}
+
+static u16_t isl29125_readBlue(struct device *i2c_dev)
+{
+	return isl29125_read16(i2c_dev, ISL29125_BLUE_L);
+}
+
+static u8_t isl29125_readStatus(struct device *i2c_dev)
+{
+	return isl29125_read8(i2c_dev, ISL29125_STATUS);
+}
+
+static int isl29125_reset(struct device *i2c_dev)
+{
+	u8_t data = 0;
+
+  /* reset to defaults */
+	isl29125_write8(i2c_dev, ISL29125_DEVICE_ID, 0x46);
+
+	/* Check reset */
+	data = isl29125_read8(i2c_dev, ISL29125_CONFIG_1);
+	data |= isl29125_read8(i2c_dev, ISL29125_CONFIG_2);
+	data |= isl29125_read8(i2c_dev, ISL29125_CONFIG_3);
+	data |= isl29125_read8(i2c_dev, ISL29125_STATUS);
+
+	return data != 0 ? -EIO : 0;
+}
+
+static int isl29125_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct isl29125_data *drv_data = dev->driver_data;
+
+	if (chan != SENSOR_CHAN_ALL &&
+	    chan != SENSOR_CHAN_RED &&
+	    chan != SENSOR_CHAN_GREEN &&
+	    chan != SENSOR_CHAN_BLUE) {
+		return -EINVAL;
+	}
+
+	if (chan == SENSOR_CHAN_RED || chan == SENSOR_CHAN_ALL) {
+		drv_data->r = isl29125_readRed(drv_data->i2c);
+	}
+	if (chan == SENSOR_CHAN_GREEN || chan == SENSOR_CHAN_ALL) {
+		drv_data->g = isl29125_readGreen(drv_data->i2c);
+	}
+	if (chan == SENSOR_CHAN_BLUE || chan == SENSOR_CHAN_ALL) {
+		drv_data->b = isl29125_readBlue(drv_data->i2c);
+	}
+	LOG_DBG("rgb: %u, %u, %u", drv_data->r, drv_data->g, drv_data->b);
+	return 0;
+}
+
+static int isl29125_channel_get(struct device *dev, enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	struct isl29125_data *drv_data = dev->driver_data;
+
+	val->val2 = 0;
+	switch (chan) {
+	case SENSOR_CHAN_RED:
+		val->val1 = drv_data->r;
+		break;
+
+	case SENSOR_CHAN_GREEN:
+		val->val1 = drv_data->g;
+		break;
+
+	case SENSOR_CHAN_BLUE:
+		val->val1 = drv_data->b;
+		break;
+
+	default:
+		val->val1 = 0;
+		break;
+	}
+
+	return 0;
+}
+
+static int isl29125_config(struct device *dev, u8_t config1, u8_t config2,
+			   u8_t config3)
+{
+	/* Set configuration registers */
+	isl29125_write8(dev, ISL29125_CONFIG_1, config1);
+	isl29125_write8(dev, ISL29125_CONFIG_2, config2);
+	isl29125_write8(dev, ISL29125_CONFIG_3, config3);
+
+	/* Check if configurations were set correctly */
+	if (isl29125_read8(dev, ISL29125_CONFIG_1) != config1 ||
+      isl29125_read8(dev, ISL29125_CONFIG_2) != config2 ||
+      isl29125_read8(dev, ISL29125_CONFIG_3) != config3) {
```

Comment:
Please replace the leading spaces with a tab

Reply:
already done in commits 385008fdd74 & 385008fdd74